libvast/vast/type.hpp

```diff
@@ -942,6 +1001,14 @@ struct sum_type_access<vast::type> {
     return static_cast<const T&>(*x);
   }
 
+  template <class T, int Pos>
+  static T* get_if(vast::type* x, sum_type_token<T, Pos>) {
```

Comment:
Can we be sure that this does not get called a lot by accident?

Reply:
No.

Reply:
I added a print statement and it seems that this is currently only called from the type resolver in the schema parser.